test/__snapshots__/github.test.ts.snap

```diff
@@ -235,12 +212,29 @@ jobs:
           role-duration-seconds: 1800
           role-to-assume: arn:aws:iam::111111111111:role/cdk-hnb659fds-deploy-role-111111111111-us-east-1
           role-external-id: Pipeline
-      - uses: aws-actions/aws-cloudformation-github-deploy@v1
+      - id: StageAFunctionStackD42C27B8
+        uses: aws-actions/aws-cloudformation-github-deploy@v1
         with:
           name: StageA-FunctionStack
-          template: https://cdk-hnb659fds-assets-111111111111-us-east-1.s3.us-east-1.amazonaws.com/c6eb397e9674810c84c295ca34efc7dd668cd429b6b9ae23d55951827cb0b25e.json
+          template: https://cdk-hnb659fds-assets-111111111111-us-east-1.s3.us-east-1.amazonaws.com/78a7673e89f59d0a6a53ec86baeb411cef605a4e970616398035dacb4bbd9ca0.json
           no-fail-on-empty-changeset: \\"1\\"
           role-arn: arn:aws:iam::111111111111:role/cdk-hnb659fds-cfn-exec-role-111111111111-us-east-1
+    outputs:
+      myout: \${{ steps.StageAFunctionStackD42C27B8.outputs.myout }}
+  StageA-Post:
+    name: Post
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+    needs:
+      - StageA-FunctionStack-Deploy
+      - StageA-BucketStack-Deploy
+      - StageA-FunctionStack-Deploy
```

Comment:
If someone can figure out why I depend on `StageA-FunctionStack-Deploy` twice, that would be great :).

Reply:
This looks to be a bug on the pipelines construct, and does not impact the performance (just looks ugly)

Reply:
Tracking: https://github.com/aws/aws-cdk/pull/18450